api/types/registry/registry.go

```diff
@@ -102,3 +105,14 @@ type SearchResults struct {
 	// Results is a slice containing the actual results for the search
 	Results []SearchResult `json:"results"`
 }
+
+// ManifestInspect describes the result obtained from contacting the
+// registry to retrieve image metadata
+// TODO(nishanttotla): Decide which other fields to add here
+type ManifestInspect struct {
+	// Digest is the content addressable digest for the image on the registry
+	Digest digest.Digest
```

Comment:
Yeah, that's basically what I had in mind. Although maybe `Manifest` should be `[]byte` so that we aren't introducing distribution types into the API, and also so that the digest can be verified. It's ugly to put JSON inside JSON though.